lib.nix

```diff
@@ -13,29 +29,47 @@ let
       (
         if (value.type or "" == "derivation")
         then
-          {
-            "${typeField}" = "nixDerivation";
-            inherit
-              (value)
-              drvPath
-              outputName
-              ;
-            outputPath = value.outPath;
+          valueWithContext (
+            {
+              "${typeField}" = "nixDerivation";
+              inherit
+                (value)
+                drvPath
+                outputName
+                ;
+              outputPath = value.outPath;
+            }
+            // (
+              if value ? version
+              then {inherit (value) version;}
+              else {}
+            )
+          ) {
+            ${builtins.unsafeDiscardStringContext value.drvPath} = value;
```

Comment:
We need the key here to also include the output name (so that `gcc.out` and `gcc.dev` are represented differently). Maybe represent the context as `{ drvPath = { outputName = { value; }; }` and use `lib.recursiveUpdate` to merge them

Reply:
I've added this, but now I'm not sure if I should have. We're accessing them as `context.${value.drvPath}.${value.outputName or "out"}`, so even if `gcc.dev` overrides `gcc.out`, we'd still access it as `gcc.dev.out` which is supposed to be the same as `gcc.out`.
That said, making this explicit removes such confusion and is probably safer for the future.